src/lib/utils.ts

```diff
@@ -65,3 +65,21 @@ export const throttledForEach = async <T, U>(
 	// We return the results of the iterator so the caller can await them as necessary
 	return await Promise.all(promises);
 };
+
+export const withRetries = async <T>(
+	func: () => Promise<T>,
+	delayDuration = 2000,
+	retries = 3,
+	err?: Error,
+): Promise<T> => {
+	if (retries <= 0) {
```

Comment:
How about moving this inside the `catch` so that we don't need to pass the errors as an argument?

Reply:
Good call, I should have seen that 😄